### PR TITLE
tx no funds fix

### PR DESCRIPTION
**Pull request by F4ever**

### CODE REVIEW

File: src/web3py/extensions/tx_utils.py [hunk 1]
```diff
@@ -42,6 +42,9 @@ def _check_transaction(transaction, params: TxParams) -> bool:
         except ContractLogicError as error:
             logger.warning({"msg": "Transaction reverted.", "error": str(error)})
             return False
+        except ValueError as error:
+            logger.error({"msg": "Not enough funds.", "error": str(error)})
```
Comment (avsetsin) [inline src/web3py/extensions/tx_utils.py:46]: is this the only possible cause of the ValueError?

File: src/web3py/extensions/tx_utils.py [hunk 2]
```diff
@@ -79,7 +82,10 @@ def _estimate_gas(transaction: ContractFunction, account: LocalAccount) -> Optio
         try:
             gas = transaction.estimate_gas({'from': account.address})
         except ContractLogicError as error:
-            logger.warning({'msg': 'Contract logic error', 'error': str(error)})
+            logger.warning({'msg': 'Contract logic error.', 'error': str(error)})
+            return None
+        except ValueError as error:
+            logger.warning({'msg': 'Execution reverted.', 'error': str(error)})
```
Comment (avsetsin) [inline src/web3py/extensions/tx_utils.py:88]: is this the only possible cause of the ValueError?

File: src/web3py/extensions/tx_utils.py (hunk 1 above)
Comment (F4ever) [reply src/web3py/extensions/tx_utils.py:46]: We will see details in "error" and if it is another exception we will see it
But for now as I know yes

File: src/web3py/extensions/tx_utils.py (hunk 2 above)
Comment (F4ever) [reply src/web3py/extensions/tx_utils.py:88]: We will see details in "error" and if it is another exception we will see it
But for now as I know yes

**Pull request merged**